libvast/CMakeLists.txt

```diff
@@ -217,14 +217,16 @@ else ()
     add_library(CAF::core ALIAS libcaf_core_${_linkage_suffix})
     target_compile_features(libcaf_core_${_linkage_suffix} INTERFACE cxx_std_17)
     target_compile_options(
-      libcaf_core_${_linkage_suffix} PRIVATE -Wno-maybe-uninitialized
-                                             -Wno-unknown-warning-option)
+      libcaf_core_${_linkage_suffix}
+      PRIVATE -Wno-maybe-uninitialized -Wno-unqualified-std-cast-call
+              -Wno-unknown-warning-option)
     VASTSystemizeTarget(libcaf_io_${_linkage_suffix})
     set_target_properties(libcaf_io_${_linkage_suffix} PROPERTIES EXPORT_NAME
                                                                   io)
     target_compile_options(
-      libcaf_io_${_linkage_suffix} PRIVATE -Wno-maybe-uninitialized
-                                           -Wno-unknown-warning-option)
+      libcaf_io_${_linkage_suffix}
+      PRIVATE -Wno-maybe-uninitialized -Wno-unqualified-std-cast-call
```

Comment:
Note to the reviewer: CAF has some unqualified calls to `std::move` via ADL, and that now triggers a warning (and rightfully so!), but for the bundled CAF it makes a lot more sense to just silence the warning.